src/main/java/bio/terra/workspace/service/workspace/WorkspaceCloudContext.java

```diff
@@ -1,24 +1,42 @@
 package bio.terra.workspace.service.workspace;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 import com.google.auto.value.AutoValue;
 import java.util.Optional;
 
 /**
  * The cloud contexts associated with a resource.
  *
+ * <p>This uses AutoValue's builder pattern to ensure it's serializable with Jackson. Otherwise,
+ * Jackson doesn't understand how to deserialize AutoValue classes.
+ *
  * <p>A workspace can have resources with each cloud provider, its "cloud context."
  * <li>For GCP, a Google Project is associated with every resource.
  */
 @AutoValue
+@JsonSerialize(as = WorkspaceCloudContext.class)
+@JsonDeserialize(builder = AutoValue_WorkspaceCloudContext.Builder.class)
 public abstract class WorkspaceCloudContext {
   /** The Google Project id for workspaces with Google context. */
+  @JsonProperty("googleProjectId")
   public abstract Optional<String> googleProjectId();
 
-  public static WorkspaceCloudContext createGoogleContext(String projectId) {
-    return new AutoValue_WorkspaceCloudContext(Optional.of(projectId));
+  // Convenience method for quickly creating an empty WorkspaceCloudContext.
+  public static WorkspaceCloudContext none() {
+    return builder().googleProjectId(Optional.empty()).build();
```

Comment:
This looks wrong to me. I get that when we have different contexts, the googleProjectId can be null. However, we shouldn't use an Optional to hold the null in the class. Optional is intended for use in return values.

Reply:
That makes sense. I don't think there's a native way to store a String but return an optional in AutoValue though, so I removed the optional everywhere and added `@Nullable`.